lib/moonshot/artifact_repository/s3_bucket_via_github_releases.rb

```diff
@@ -70,13 +77,143 @@ def transfer_release_asset_to_s3(version)
     def github_to_s3(version, s3_name)
       Dir.mktmpdir('github_to_s3', Dir.getwd) do |tmpdir|
         Dir.chdir(tmpdir) do
-          sh_out("hub release download #{version}")
-          file = Dir.glob("*#{version}*.tar.gz").fetch(0)
+          file = download_from_github(version)
           upload_to_s3(file, s3_name)
         end
       end
     end
 
+    # Uploads the file to s3 and verifies the checksum.
+    #
+    # @param file [String] File to be uploaded to s3.
+    # @param key [String] Name of the object to be created on s3.
+    # @raise [RuntimeError] If the file fails to upload correctly after 3
```

Comment:
I don't believe this is a correct use of [RuntimeError](https://ruby-doc.org/core-2.2.0/RuntimeError.html)

See also: [RuntimeException](https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html)

Note: the implementations are almost all things you messed up, you should have written your program differently to avoid this situation.

/pedant